### PR TITLE
Use Formula#oldnames to silence deprecation notices

### LINKED ISSUE #1223: bundle dump spams warning messages to console regarding a deprecated function call
When running `$ brew bundle dump --force` I get this error message. Which suggests I report it here.

```
Warning: Calling Formula#oldname is deprecated! Use Formula#oldnames instead.
Please report this issue to the homebrew/bundle tap (not Homebrew/brew or Homebrew/homebrew-core), or even better, submit a PR to fix it:
  /opt/homebrew/Library/Taps/homebrew/homebrew-bundle/lib/bundle/brew_dumper.rb:160
```

### PR DESCRIPTION
**Pull request by colindean**
This doesn't really fully safely use `Formula#oldnames`, just
effectively uses what `Formula#oldname` does now.

https://github.com/Homebrew/brew/blob/4bbc0af6935816c75e0057364a84fd7fa8de071b/Library/Homebrew/formula.rb#L502-L515

Fixes #1223

### CODE REVIEW

**Comment by colindean** [comment]
Test failure looks like a temporary failure to retrieve the latest https://formulae.brew.sh/api/cask.jws.json.

File: spec/bundle/brew_dumper_spec.rb [hunk 1]
```diff
@@ -14,6 +14,7 @@
                     name:                   "foo",
                     desc:                   "foobar",
                     oldname:                "oldfoo",
+                    oldnames:               ["oldfoo"],
```
Comment (vraravam) [inline spec/bundle/brew_dumper_spec.rb:17]: Shouldn't the previous line also be deleted in this case?

File: spec/bundle/brew_dumper_spec.rb [hunk 2]
```diff
@@ -58,6 +59,7 @@
                     name:                   "bar",
                     desc:                   "barfoo",
                     oldname:                nil,
+                    oldnames:               [],
```
Comment (vraravam) [inline spec/bundle/brew_dumper_spec.rb:62]: Shouldn't the previous line also be deleted in this case?

File: spec/bundle/brew_dumper_spec.rb [hunk 3]
```diff
@@ -119,6 +121,7 @@
                     name:                   "baz",
                     desc:                   "",
                     oldname:                nil,
+                    oldnames:               [],
```
Comment (vraravam) [inline spec/bundle/brew_dumper_spec.rb:124]: Shouldn't the previous line also be deleted in this case?

File: spec/bundle/brew_dumper_spec.rb (hunk 1 above)
Comment (colindean) [reply spec/bundle/brew_dumper_spec.rb:17]: It could, but the method is still present on the class this stub stubs.

**Review by jacobbednarz — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks for workaround here @colindean and for merging @jacobbednarz!

**Pull request merged**